### PR TITLE
Initial run_action_instance command

### PR DESCRIPTION
**Pull request by m2bright**

### CODE REVIEW

File: rapid/client/communicator/client_communicator.py [hunk 1]
```diff
@@ -175,6 +179,16 @@ def _string_header_values(self, headers):
             pass
         return _headers
 
+    def _default_get(self, uri):
+        headers = {}
+        master_key = StoreService.get_master_key(self.flask_app)
+        headers = self._string_header_values(headers)
+
+        if master_key is not None:
+            headers['X-Rapidci-Api-Key'] = master_key
```
Comment (BambooBryson) [inline rapid/client/communicator/client_communicator.py]: clever!

File: rapid/client/action_instance_runner.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    def __init__(self, rapid_config):
+        # type: (ClientConfiguration) -> None
+        self.client_config = rapid_config
+
+    def run_action_instance(self, app, action_instance_id):
+        # type: (Flask, int) -> None
+        communicator = ClientCommunicator(self.client_config.master_uri, self.client_config.quarantine_directory, app, self.client_config.verify_certs)
+        communicator.register(self.client_config)
+        request_json = communicator.get_work_request_by_action_instance_id(action_instance_id)
+        executor = Executor(WorkRequest(request_json),
```
Comment (jlesueur) [inline rapid/client/action_instance_runner.py]: won't fully understand all this until I see executor (I assume it's later in the PR).

File: rapid/client/executor.py [hunk 3]
```diff
@@ -95,14 +95,17 @@ def _create_logger(self):
...
         """
-        self.thread = threading.Thread(target=self._start_child_process)
-        self.thread.daemon = True
-        self.thread.start()
+        if threaded:
+            self.thread = threading.Thread(target=self._start_child_process)
+            self.thread.daemon = True
+            self.thread.start()
+        else:
+            self._start_child_process()
```
Comment (jlesueur) [inline rapid/client/executor.py]: docker instances run inline, not threaded. :+1:

File: rapid/lib/work_request.py [hunk 4]
```diff
@@ -56,19 +56,23 @@ def prepare_to_send(self, data_in):
...
-                    tmp = attr
-                    if tmp == '_grain':
-                        tmp = 'grain'
-                    setattr(self, attr, data_in[tmp])
+                    if attr == '_grain':
+                        for g_attr in ['_grain', 'grain']:
+                            if g_attr in data_in:
+                                setattr(self, '_grain', data_in[g_attr])
+                    else:
+                        setattr(self, attr, data_in[attr])
```
Comment (jlesueur) [inline rapid/lib/work_request.py]: getting prepared for an attribute rename? `_grain` and `grain`?

Comment (m2bright) [reply rapid/lib/work_request.py]: No, this is backward compatibility that wasn't working.

**Pull request merged**